src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -29,9 +29,32 @@ import { TERMINAL_COMMAND_ID } from 'vs/workbench/parts/terminal/common/terminal
 import { Command } from 'vs/editor/browser/editorExtensions';
 import { timeout } from 'vs/base/common/async';
 import { FindReplaceState } from 'vs/editor/contrib/find/findState';
+import { execSync } from 'child_process';
+import { isWindows } from 'vs/base/common/platform';
 
 export const TERMINAL_PICKER_PREFIX = 'term ';
 
+function getCwd(configHelper: ITerminalConfigHelper, activeInstance: ITerminalInstance): string {
+	switch (configHelper.config.splitCwdSource) {
+		case 'ShellDefault': {
+			// allow default behavior
+			return '';
+		}
+		case 'SourceInitialCwd': {
+			return activeInstance.getInitialCwd();
+		}
+		case 'Cwd': {
+			if (!isWindows) {
+				let pid = activeInstance.processId;
+				let output = execSync('lsof -p ' + pid + ' | grep cwd').toString();
```

Comment:
We should make this async, I think we're working on promises in all the places getCwd is used.